effect/shared/src/main/scala/scalaz/effect/Async.scala

```diff
@@ -1,29 +1,27 @@
 // Copyright (C) 2017-2018 John A. De Goes. All rights reserved.
 package scalaz.effect
 
+import scalaz.Void
+
 /**
  * The `Async` class describes the return value of an asynchronous effect
  * that is imported into an `IO` value.
  *
  * Asynchronous effects can return `later`, which represents an uninterruptible
  * asynchronous action, `now` which represents a synchronously computed value,
- * or `maybeLater`, which represents an interruptible asynchronous action.
+ * `maybeLater`, which represents an interruptible asynchronous action or `maybeLaterIO`
+ * which represents an interruptible asynchronous action where the canceler has the
+ * form `Throwable => IO[Void, Unit]`
  */
 sealed abstract class Async[E, A]
 object Async {
-  type Interruptor = Throwable => Unit
+  type Canceler     = Throwable => Unit
+  type PureCanceler = Throwable => IO[Void, Unit]
```

Comment:
These are good. Now create some `val` for these:

```scala
val Canceler: Canceler = _ => ()
val PureCanceler: PureCanceler = _ => IO.unit[Void]
```

Reply:
added. Thanks 👍